docs/weaviate/concepts/data-import.mdx

```diff
@@ -0,0 +1,71 @@
+---
+title: Data import
+sidebar_position: 11
+description: "Theoretical explanation of client-side and server-side batch imports."
+image: og/docs/concepts.jpg
+---
+
+Weaviate offers two flexible methods for importing data in bulk: **client-side batching** and **server-side batching**. This allows you to choose the best strategy based on your specific needs.
+
+- **Client-side batching** <br/>
+  In the traditional client-side approach, **your application is responsible for grouping data into batches**. You define the exact size of each batch (e.g., 100 objects) using the appropriate [client library method](../manage-objects/import.mdx). The client then sends these pre-defined chunks to the Weaviate server.
+
+  This method gives you direct control over the import process but may require manual tuning of parameters like `batch_size` and `concurrent_requests` to achieve optimal performance and avoid overwhelming the server.
+
+- **Server-side batching** <br/>
+  Server-side batching, or **automatic mode**, is a more robust and the recommended approach. Here, the client sends data as a continuous stream, and the **Weaviate server manages the data flow**.
+
+  Using an internal queue and a dynamic _[backpressure](https://en.wikipedia.org/wiki/Backpressure_routing)_ mechanism, the server tells the client how much data to send next based on its current workload. This simplifies your client code, eliminates the need for manual tuning, and results in a more efficient and resilient data import process.
+
+:::tip
+
+For **code examples**, check out the [How-to: Batch import](../manage-objects/import.mdx) guide. Currently, only the Python client supports batch imports.
+
+:::
+
+---
+
+## Server-side batching
+
+:::caution Technical preview
+
+Server-side batching was added in **`v1.33`** as a **technical preview**.<br/><br/>
+This means that the feature is still under development and may change in future releases, including potential breaking changes.
+**We do not recommend using this feature in production environments at this time.**
+
+:::
+
+Weaviate's server-side batching, also known as **automatic batching**, is an intelligent mechanism designed to make data ingestion simpler, faster, and more robust. Instead of manually tuning batch parameters on the client side, you can let the server dynamically manage the data flow for optimal performance.
+
+This mode is a drop-in replacement for `fixed_size` or `dynamic` batching and is the recommended method for importing data.
+
+### How it works
+
+When you use the `automatic()` batching context manager, you initiate a persistent connection to the server for the duration of the batch job.
```

Comment:
This language here is quite Python-centric (i.e. `automatic()` and `context manager`. Wdyt about something like:

```
When an automatic batch import is initiated, the client opens a persistent connection to the server for the duration of the batch job.
```